examples/erc20-counter/contracts/test/Counter.t.sol

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.20;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+import {Receipt as RiscZeroReceipt} from "risc0/IRiscZeroVerifier.sol";
+import {RiscZeroMockVerifier} from "risc0/test/RiscZeroMockVerifier.sol";
+import {Counter} from "../src/Counter.sol";
+import {Steel, Beacon, Encoding} from "risc0/steel/Steel.sol";
+import {ERC20} from "openzeppelin-contracts/token/ERC20/ERC20.sol";
+
+contract ERC20FixedSupply is ERC20 {
+    constructor(string memory name, string memory symbol, address owner) ERC20(name, symbol) {
+        _mint(owner, 1000);
+    }
+}
+
+contract CounterTest is Test {
```

Comment:
These tests are actually never run in the CI. Should we change that or at least move the tests of the commitment validation into the general risc0-ethereum foundry tests

Reply:
I'd like to have these tests running in CI if at all possible. Does this run locally in Anvil alright, considering the use of `Beacon.blockRoot`?

Reply:
The beacon commitment test relies on anvil fork mode and thus an RPC endpoint. However, using a publicnode.com I believe would be fine, even in the CI.

Reply:
I'm happy to give it a try using publicnode. Hopefully it'll be reliable and that'll just solve the issue 👌